components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -3384,43 +3404,38 @@ private void associateAuthenticationHistory(SessionDataCacheEntry resultFromLogi
-     *
-     * @param cookie
+     * Returns the SessionContext associated with the cookie value, if there is a one.
+     * @param cookieValue String value of the cookie of commonAuthId.
      * @param loginTenantDomain Login tenant domain.
-     * @return the associate SessionContext or null.
+     * @return he associate SessionContext or null.
      */
-    private SessionContext getSessionContext(Cookie cookie, String loginTenantDomain) {
+    private SessionContext getSessionContext(String cookieValue, String loginTenantDomain) {
 
-        if (cookie != null) {
-            String sessionContextKey = DigestUtils.sha256Hex(cookie.getValue());
+        if (StringUtils.isNotBlank(cookieValue)) {
+            String sessionContextKey = DigestUtils.sha256Hex(cookieValue);
             return FrameworkUtils.getSessionContextFromCache(sessionContextKey, loginTenantDomain);
         }
         return null;
     }
 
     /**
-     * Gets the last authenticated value from the commonAuthId cookie
+     * Gets the last authenticated value from the commonAuthId cookie value.
      *
-     * @param cookie CommonAuthId cookie
-     * @param loginTenantDomain Login tenant domain
-     * @return the last authenticated timestamp
+     * @param cookieValue       String CommonAuthId cookie values.
+     * @param loginTenantDomain String Login tenant domain.
+     * @return long The last authenticated timestamp.
      */
-    private long getAuthenticatedTimeFromCommonAuthCookie(Cookie cookie, String loginTenantDomain) {
+    private long getAuthenticatedTimeFromCommonAuthCookieValue(String cookieValue, String loginTenantDomain) {
 
         long authTime = 0;
-        if (cookie != null) {
-            String sessionContextKey = DigestUtils.sha256Hex(cookie.getValue());
-            SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(sessionContextKey,
-                    loginTenantDomain);
-            if (sessionContext != null) {
-                if (sessionContext.getProperty(FrameworkConstants.UPDATED_TIMESTAMP) != null) {
-                    authTime = Long.parseLong(
-                            sessionContext.getProperty(FrameworkConstants.UPDATED_TIMESTAMP).toString());
-                } else {
-                    authTime = Long.parseLong(
-                            sessionContext.getProperty(FrameworkConstants.CREATED_TIMESTAMP).toString());
-                }
+        SessionContext sessionContext = getSessionContext(cookieValue, loginTenantDomain);
+        if (sessionContext != null) {
+            if (sessionContext.getProperty(FrameworkConstants.UPDATED_TIMESTAMP) != null) {
+                authTime = Long.parseLong(
+                        sessionContext.getProperty(FrameworkConstants.UPDATED_TIMESTAMP).toString());
+            } else {
+                authTime = Long.parseLong(
+                        sessionContext.getProperty(FrameworkConstants.CREATED_TIMESTAMP).toString());
```

Comment:
Seems like we can improve the code shifting to the right. Also we can use if without else with return statements.

Reply:
Fixed ad3cb637ce2bd2485bc1b335e6f68be0d87ef70b